### PR TITLE
Fix several warnings

### PR DESCRIPTION
**Pull request by pdalpra**
This PR fixes a bunch of warnings, mostly deprecations from scala-reflect and a few from sbt.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by dwijnand** [comment]
LGTM. Thanks man.

File: compile/api/src/main/scala/sbt/ClassToAPI.scala [hunk 1]
```diff
@@ -26,7 +26,7 @@ object ClassToAPI {
     }
 
   // Avoiding implicit allocation.
-  private def arrayMap[T <: AnyRef, U <: AnyRef: ClassManifest](xs: Array[T])(f: T => U): Array[U] = {
```
Comment (jsuereth) [inline compile/api/src/main/scala/sbt/ClassToAPI.scala:29]: Ooh, we're finally not compiling to Scala 2.9 :)

**Comment by jsuereth** [comment]
Thanks again for the cleanups!

**Comment by pdalpra** [comment]
My pleasure :)

**Comment by eed3si9n** [comment]
This is awesome stuff.

**Pull request merged**